File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    if s[0] == ')':
        return False
    elif s[-1] == '(':
        return False
    else:
        count = 0
        for ps in s:
            if ps == '(':
                count+=1
            elif ps == ')':
                count -=1
            if count < 0:
                return False
        return True if count == 0 else False

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        if (s.charAt(0) == ')' || s.charAt(s.length() - 1) == '(') return false;
        int count = 0;
        for(char ch : s.toCharArray()){
            if(ch == '(') count++;
            else if(ch == ')') count--;
            
            if(count < 0) return false;
        }
        
        return count == 0;
    }
}